src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -710,11 +715,12 @@ public TaskResource updateNotes(String taskId, NotesRequest notesRequest) {
             throw new RoleAssignmentVerificationException(ROLE_ASSIGNMENT_VERIFICATIONS_FAILED);
         }
 
-        return taskResourceQueryResult.get().getTaskRoleResources().stream().map(
-            cftTaskMapper::mapToTaskRolePermissions)
+        return taskResourceQueryResult.get().getTaskRoleResources().stream()
+            .filter(taskRoleResource -> taskRoleResource.getRead().equals(Boolean.TRUE))
```

Comment:
Similar to other endpoints ,we retrieve role assignments for the logged in user. 
1. We compare logged in user role assignments against the Task Roles, if they match and has Read permissions to any of the roles we retrieve the taskResource object. Since the taskResource and TaskRoleResource has 1-many relationship, the TaskResource record gives all the roles so I have to filter based on the read permission.

to avoid that filter, we need to return Collection of TaskRoleResources which has only read permissions in it.
In my opinion, there is no difference in retrieving From TaskResoure or TaskRoleResource its the same.

Reply:
Let's have a quick chat about this one.